integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/LogicalRelationVisitor.java

```diff
@@ -0,0 +1,142 @@
+package marquez.spark.agent.lifecycle.plan;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import marquez.spark.agent.client.LineageEvent.Dataset;
+import marquez.spark.agent.client.LineageEvent.DatasetFacet;
+import org.apache.spark.SparkContext;
+import org.apache.spark.sql.catalyst.catalog.CatalogTable;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.execution.datasources.HadoopFsRelation;
+import org.apache.spark.sql.execution.datasources.LogicalRelation;
+import org.apache.spark.sql.execution.datasources.jdbc.JDBCOptions;
+import org.apache.spark.sql.execution.datasources.jdbc.JDBCRelation;
+import scala.collection.JavaConversions;
+import scala.runtime.AbstractFunction0;
+import scala.runtime.AbstractPartialFunction;
+
+/**
+ * {@link LogicalPlan} visitor that attempts to extract a {@link Dataset} from a {@link
+ * LogicalRelation}. The {@link org.apache.spark.sql.sources.BaseRelation} is tested for known
+ * types, such as {@link HadoopFsRelation} or {@link JDBCRelation}s, as those are easy to extract
+ * exact dataset information.
+ *
+ * <p>For {@link HadoopFsRelation}s, it is assumed that a single directory maps to a single {@link
+ * Dataset}. Any files referenced are replaced by their parent directory and all files in a given
+ * directory are assumed to belong to the same {@link Dataset}. Directory partitioning is currently
+ * not addressed.
+ *
+ * <p>For {@link JDBCRelation}s, {@link * Dataset} naming expects the namespace to be the JDBC
```

Comment:
_Typo:_ Drop the `*` in `{@link * Dataset}`?

Reply:
👍🏽